src/services/scriptingservice.cpp

```diff
@@ -496,6 +496,26 @@ QVariant ScriptingService::callNoteTaggingByObjectHook(
     return QVariant();
 }
 
+void ScriptingService::callWindowStateChangeHook(const QString &windowStateStr)
```

Comment:
@fnkabit, @Waqar144 Do you think there ever will be a "StateChangeHook" for another windows than the MainWindow? Would that make sense?

Reply:
I think no. I like the idea that we are single window only. A single window is giving you all the abilities. But it also depends on what the user has in mind.

Reply:
Ok, thanks.